src/components/dropdown/dropdown.tsx

```diff
@@ -0,0 +1,192 @@
+import { DropdownStyle, OptionGroupStyle, OptionStyle } from './styles';
+import { SizeType } from '../../core/types/size';
+import IonIcon from '../icons';
+import React, { useState } from 'react';
+
+type SelectSizes = Extract<SizeType, 'sm' | 'md'>;
+
+export interface OptionProps {
+  disabled?: boolean;
+  label: string;
+  selected?: boolean;
+  value: string | number;
+}
+
+export interface OptionGroupProps {
+  disabled?: boolean;
+  options: OptionProps[];
+  label: string;
+}
+
+export interface DropdownProps {
+  disabled?: boolean;
+  multiple?: boolean;
+  options?: OptionProps[];
+  optionsGroup?: OptionGroupProps[];
+  selectedOptions?: OptionProps[];
+  size?: SelectSizes;
+  onSelectedOption: Function;
+}
+
+export interface OptionsConfig extends DropdownProps {
+  indexGroup?: number;
+}
+
+const isDisabled = (option: OptionProps, disabled?: boolean): boolean => {
+  if (disabled || option.disabled) return true;
+  return false;
+};
+const isSelectingNotMultiple = (
+  option: OptionProps,
+  multiple?: boolean
+): boolean => {
+  if (!option.selected && !multiple) return true;
+  return false;
+};
+
+const Options = ({
+  disabled,
```

Comment:
```suggestion
  disabled = false,
```
Optional props can be initialized with a default value